hedera-node/src/main/java/com/hedera/services/context/ImmutableStateChildren.java

```diff
@@ -51,7 +52,8 @@ public class ImmutableStateChildren implements StateChildren {
     private final WeakReference<MerkleMap<EntityNum, MerkleAccount>> accounts;
     private final WeakReference<MerkleMap<EntityNum, MerkleTopic>> topics;
     private final WeakReference<MerkleMap<EntityNum, MerkleToken>> tokens;
-    private final WeakReference<MerkleMap<EntityNumPair, MerkleUniqueToken>> uniqueTokens;
+    // UniqueTokenMapAdapter is constructed on demand, so a strong reference needs to be held.
```

Comment:
I'm having trouble grasping this comment. 😝 

What happens if this is a `WeakReference`?

Reply:
If it's a weak reference, then there would be no one holding on to the reference constructed by:
`UniqueTokenAdapter.wrap(token)` that gets provided to the instance when set.

Reply:
Does this mean it's actually safe to drop all the `WeakReference`'s here?